src/app/components/RouterTabs/index.tsx

```diff
@@ -10,28 +10,13 @@ type RouterTabsProps = {
   }[]
 }
 
-function useRouteMatch(patterns: string[]) {
-  const { pathname } = useLocation()
-
-  for (let i = 0; i < patterns.length; i += 1) {
-    const pattern = patterns[i]
-    const possibleMatch = matchPath(pattern, pathname)
-    if (possibleMatch !== null) {
-      return possibleMatch
-    }
-  }
-
-  return null
-}
-
 export const RouterTabs: FC<RouterTabsProps> = ({ tabs }) => {
-  const tabsPaths = tabs.flatMap(num => num.to)
-  const routeMatch = useRouteMatch(tabsPaths)
-  const currentTab = routeMatch?.pattern?.path
+  const { pathname } = useLocation()
+  const currentTab = tabs.find(tab => matchPath(tab.to, pathname))
```

Comment:
Does this take card of the reason why flattening was used before? (in `tabs.flatMap`).

Reply:
Looked like a mistake to me

Reply:
Maybe, maybe not. @buberdds wrote the previous version of the code, maybe he can explain the reason?

Reply:
refactor looks Ok